### DISCUSSION TITLE
Build VSCode extension for live preview

### DISCUSSION THREAD

**Discussion by zenorocha** [resend/react-email#574] [Ideas]
Inspiration:
* https://twitter.com/chriscoyier/status/1572644485628305409

Reference:
* https://marketplace.visualstudio.com/items?itemName=tht13.html-preview-vscode
* https://marketplace.visualstudio.com/items?itemName=DanielKnights.vscode-mjml

**Comment by EmersonGarrido** [comment]
Plus exemple : https://previewjs.com

**Comment by lucianodiisouza** [comment]
To be honest, I think that I have zero ability to build it, but I want to challenge myself 🤣

looking at the @EmersonGarrido  example, this opened issue continues valid @zenorocha  ?

**Comment by EmersonGarrido** [comment]
@lucianodiisouza It's open, I started to build something and I also have an example of an html preview extension in my open github.

**Comment by m-shaka** [comment]
Any updates? Please let me share my short survey.
I tried [@previewjs/cli](https://previewjs.com/docs/platforms/cli) to preview react-email components and it seems to work well, even though some errors are reported partly because an `html` tag is in another `html` tag.

And Preview.js is publishing its packages in npm so perhaps they can be used for react-email's extension.
https://www.npmjs.com/search?q=%40previewjs

**Comment by BrunoPierca** [comment]
Hi! is this still a WIP? I'm interested in adding a live preview in my own site showing how the emails would look, I'm also thinking about creating something like a drag and drop email creator.

I'm having issues rendering some styles and I thought I might find something to fix that in this kind of feature.

Thank you

**Comment by cungminh2710** [comment]
@fwouts Hi François, any thoughts on adding better supports for `react-email`? Or workaround we can do in the meantime 😃

**Comment by Abdo-reda** [comment]
I worked on an extension specifically for react email, it would be nice if anyone can check it out and give me feedback 🙌 [github repo](https://github.com/Abdo-reda/react-email-vscode-extension)
